frontend/src/components/Status.tsx

```diff
@@ -42,18 +42,25 @@ export default function Status({
 }: {status: string, cluster?: any}) {
   const failedStatuses = new Set(['CREATE_FAILED', 'DELETE_FAILED', 'UPDATE_FAILED']);
 
-  const statusMap: Record<string, StatusIndicatorProps.Type> = {"CREATE_IN_PROGRESS": 'in-progress',
-    "CREATE_COMPLETE": 'success',
-    "CREATE_FAILED": 'error',
-    "DELETE_IN_PROGRESS": 'in-progress',
-    "DELETE_FAILED": 'error',
-    "RUNNING": 'success',
-    "STOPPED": 'error',
-    "STOP_REQUESTED": 'in-progress',
-    "UPDATE_FAILED": 'error',
-    "UPDATE_IN_PROGRESS": 'in-progress',
-    "UNKNOWN": 'error',
-    "UPDATE_COMPLETE": 'success'};
+  const statusMap: Record<string, StatusIndicatorProps.Type> = {'CREATE_IN_PROGRESS': 'in-progress',
+    'CREATE_COMPLETE': 'success',
+    'CREATE_FAILED': 'error',
+    'CANCELLED': 'error',
+    'CONFIGURING': 'in-progress',
+    'COMPLETING': 'in-progress',
+    'COMPLETED': 'success',
+    'DELETE_IN_PROGRESS': 'in-progress',
+    'DELETE_FAILED': 'error',
+    'FAILED': 'error',
+    'PENDING': 'pending',
+    'RUNNING': 'success',
+    'STOPPED': 'error',
+    'SUCCESS': 'success',
```

Comment:
issue: I see some problems with this approach
- we are extending the statuses every time we need to handle a new resource or a new state of an existing resource, and this can go on indefinitely (I think)
- we mix different statuses of different resources in the same dictionary, and it's very difficult to understand the entity to which a particular status belongs to

While we can solve these problems in many ways, one solution I propose is to extract the mapping logic from this component and put them in entity-specific mapping functions, so that we can test the logic easily

```typescript
function instanceToStatus(instance): StatusIndicatorProps.Type {
  // map instance status to component status type
}

function jobToStatus(job): StatusIndicatorProps.Type {
  // map job status to component status type
}
...
```

Let's have a quick chat if you have better ideas on how to solve this matter!

Reply:
I ended up adding the statusMapOverrides option and leveraging it for instances which have a lower-case status.